openpype/pipeline/load/plugins.py

```diff
@@ -82,20 +81,67 @@ def apply_settings(cls, project_settings, system_settings):
                 print("  - setting `{}`: `{}`".format(option, value))
             setattr(cls, option, value)
 
+    @classmethod
+    def has_valid_extension(cls, repre_doc):
+        """Has representation document valid extension for loader.
+
+        Args:
+            repre_doc (dict[str, Any]): Representation document.
+
+        Returns:
+             bool: Representation has valid extension
+        """
+
+        if "*" in cls.extensions:
+            return True
+
+        # Get representation main file extension from 'context'
+        repre_context = repre_doc.get("context") or {}
+        ext = repre_context.get("ext")
+        if not ext:
+            # Legacy way how to get extensions
+            path = repre_doc["data"].get("path")
```

Comment:
Only open issue I think is [this comment](https://github.com/ynput/OpenPype/pull/4492#pullrequestreview-1311019748) related to I believe this line of code. Somehow `repre_doc` doesn't have `data` in some cases?

Which might need to make this:
```suggestion
            path = repre_doc.get("data", {}).get("path")
```

Reply:
It would be good however to double check whether `data` is ever available in `representation` in the `context`. It might very well be that it's a much more minimalized representation than the full document. I have no idea, but it did struck me as odd that a representation wouldn't have a `data` entry at all.

Reply:
We've checked and the representation in database has `"data"` so it must be something in the code.